app/scripts/views/mixins/password-strength-mixin.js

```diff
@@ -0,0 +1,100 @@
+  'lib/require-on-demand',
+  'lib/url',
+  'underscore'
+], function (p, requireOnDemand, Url, _) {
+  'use strict';
+
+  var PasswordStrengthMixin = {
+    initialize: function (options) {
+      this._able = options.able;
+    },
+
+    _isPasswordStrengthCheckEnabledValue: undefined,
+    isPasswordStrengthCheckEnabled: function () {
+      var self = this;
+      if (_.isUndefined(self._isPasswordStrengthCheckEnabledValue)) {
+        var abData = {
+          isMetricsEnabledValue: self.metrics.isCollectionEnabled(),
+          uniqueUserId: self.user.get('uniqueUserId'),
+          // the window parameter will override any ab testing features
+          forcePasswordStrengthCheck: Url.searchParam('passwordStrengthCheck', self.window.location.search)
+        };
+
+        self._isPasswordStrengthCheckEnabledValue =
+              self._able.choose('passwordStrengthCheckEnabled', abData);
+
+        if (self._isPasswordStrengthCheckEnabledValue) {
+          self._logStrengthExperimentEvent('enabled');
+        }
+      }
+      return self._isPasswordStrengthCheckEnabledValue;
+    },
+
+    _passwordStrengthCheckerPromise: undefined,
+    getPasswordStrengthChecker: function () {
+      // returns a promise that resolves once the library is loaded.
+      var self = this;
+      if (! self._passwordStrengthCheckerPromise) {
+        self._passwordStrengthCheckerPromise = requireOnDemand('passwordcheck')
+          // Log any failures loading the script
+          .fail(self.logError.bind(self))
+          .then(function (PasswordCheck) {
+            return new PasswordCheck();
+          });
+      }
+      return self._passwordStrengthCheckerPromise;
+    },
+
+    /**
+     * Check the password strength. Returns a promise that resolves
+     * when the check is complete. Promise resolves to `DISABLED` if
+     * password strength checker is disabled.
+     *
+     * Usage:
+     *
+     *   view.checkPasswordStrength(password)
+     *     .then(function (status) {
+     *      // do something with the status
+     *     });
+     *
+     * @method checkPasswordStrength
+     * @returns {Promise}
+     */
+    checkPasswordStrength: function (password) {
+      var self = this;
+      if (! self.isPasswordStrengthCheckEnabled()) {
+        return p('DISABLED');
+      }
+
+      return self.getPasswordStrengthChecker()
+        .then(function (passwordStrengthChecker) {
+          var deferred = p.defer();
+          passwordStrengthChecker(password, function (passwordCheckStatus) {
+            // in the future, do some fancy tooltip here.
+            passwordCheckStatus = passwordCheckStatus || 'UNKNOWN';
+
+            if (passwordCheckStatus === 'BLOOMFILTER_HIT' ||
+                passwordCheckStatus === 'BLOOMFILTER_MISS') {
+              self._logStrengthExperimentEvent('bloomfilter_used');
+            }
+            self._logStrengthExperimentEvent(passwordCheckStatus);
+
+            deferred.resolve(passwordCheckStatus);
```

Comment:
@vladikoff The point was to see if someone 'accidentally' skips the password field. Of course, this would be caught by our Signup page, and they would eventually have to enter a password. We wanted to log that, and so, this is expected behavior.